### PR TITLE
Initial feedback

### PR DESCRIPTION
**Pull request by joshmgross**
Leaving feedback in the form of a PR

Changes:
* Removed unused imports and variables
* Updated LICENSE
* Removed `shelljs` and `@types/shelljs` packages as they were not used

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "Actions Http Client",
   "main": "index.js",
   "scripts": {
-    "build": "rm -Rf ./_out && tsc && cp package*.json ./_out && cp *.md ./_out && cp LICENSE ./_out && cp actions.png ./_out",
+    "build": "tsc",
```
Comment (joshmgross) [inline package.json]: This wasn't working on my Windows machine (`rm` not found) and it looks like this is combining `build` and some release since `LICENSE` is copied over.

File: tsconfig.json [hunk 2]
```diff
@@ -8,8 +8,5 @@
         "outDir": "_out",
         "forceConsistentCasingInFileNames": true
     },
-    "files": [
-        "index.ts",
-        "auth.ts"
-    ]
+    "include": ["*.ts"]
```
Comment (joshmgross) [inline tsconfig.json]: I think this was missing `interface.ts` and `proxy.ts` when compiling

File: __tests__/auth.test.ts [hunk 3]
```diff
@@ -1,6 +1,5 @@
-import * as httpm from '../_out';
```
Comment (bryanmacfarlane) [inline __tests__/auth.test.ts]: I actually want to test the output that we are going to publish.  Not the root source code.

File: __tests__/auth.test.ts [hunk 4]
```diff
@@ -1,6 +1,5 @@
...
-import * as path from 'path';
-import * as am from '../_out/auth';
```
Comment (bryanmacfarlane) [inline __tests__/auth.test.ts]: same

File: __tests__/keepalive.test.ts [hunk 5]
```diff
@@ -1,13 +1,7 @@
-import * as httpm from '../_out';
```
Comment (bryanmacfarlane) [inline __tests__/keepalive.test.ts]: same

File: __tests__/proxy.test.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import * as pm from '../_out/proxy';
```
Comment (bryanmacfarlane) [inline __tests__/proxy.test.ts]: same

File: auth.ts [hunk 7]
```diff
@@ -11,7 +11,7 @@ export class BasicCredentialHandler implements ifm.IRequestHandler {
     }
 
     prepareRequest(options:any): void {
-        options.headers['Authorization'] = 'Basic ' + new Buffer(this.username + ':' + this.password).toString('base64');
```
Comment (bryanmacfarlane) [inline auth.ts]: good

File: index.ts [hunk 8]
```diff
@@ -70,7 +69,7 @@ export function isHttps(requestUrl: string) {
 }
 
 export class HttpClient {
-    userAgent: string | null | undefined;
```
Comment (bryanmacfarlane) [inline index.ts]: good since I made constructor arg optional

File: package.json [hunk 9]
```diff
@@ -4,7 +4,7 @@
   "description": "Actions Http Client",
   "main": "index.js",
   "scripts": {
-    "build": "rm -Rf ./_out && tsc && cp package*.json ./_out && cp *.md ./_out && cp LICENSE ./_out && cp actions.png ./_out",
```
Comment (bryanmacfarlane) [inline package.json]: we need this to build a fully realized package that we're going to publish in _out.  The alternative is a make.js which is where I started with (and why shelljs/make was added.  that's an xplat outer build solution.

File: package.json [hunk 10]
```diff
@@ -24,10 +24,7 @@
   "devDependencies": {
     "@types/jest": "^24.0.25",
     "@types/node": "^13.1.5",
-    "@types/shelljs": "^0.8.6",
     "jest": "^24.9.0",
-    "nock": "^11.7.2",
```
Comment (bryanmacfarlane) [inline package.json]: leave this for now because there's some more tests I was going to add that need nock.

File: tsconfig.json (hunk 2 above)
Comment (bryanmacfarlane) [reply tsconfig.json]: nope.  they compiled because index and auth referenced interface and proxy.

File: __tests__/auth.test.ts (hunk 3 above)
Comment (bryanmacfarlane) [reply __tests__/auth.test.ts]: this needs to be out.  see the tsconfig which builds to _out

File: __tests__/basics.test.ts [hunk 11]
```diff
@@ -1,14 +1,11 @@
-import * as httpm from '../_out';
```
Comment (bryanmacfarlane) [inline __tests__/basics.test.ts]: same

File: __tests__/auth.test.ts (hunk 3 above)
Comment (joshmgross) [reply __tests__/auth.test.ts]: I assumed that `ts-jest` would use the same output as the tsconfig so that this wouldn't be necessary

File: auth.ts (hunk 7 above)
Comment (joshmgross) [reply auth.ts]: Using `new Buffer` is deprecated, this is equivalent. See https://nodejs.org/api/buffer.html#buffer_new_buffer_array

File: index.ts (hunk 8 above)
Comment (joshmgross) [reply index.ts]: An optional parameter in typescript is `T | undefined` so the input won't be `null` if left out, it will be `undefined`.

See https://www.typescriptlang.org/docs/handbook/functions.html#optional-and-default-parameters

File: package.json (hunk 10 above)
Comment (joshmgross) [reply package.json]: Added `nock` back in

File: tsconfig.json (hunk 2 above)
Comment (joshmgross) [reply tsconfig.json]: I reverted this change

File: package.json (hunk 9 above)
Comment (joshmgross) [reply package.json]: Reverted this change, I was thinking that there would separate `build` and `release`/`publish` scripts

File: auth.ts (hunk 7 above)
Comment (bryanmacfarlane) [reply auth.ts]: Yep, as I point out above, this was a good change.

File: index.ts (hunk 8 above)
Comment (bryanmacfarlane) [reply index.ts]: OK

File: __tests__/basics.test.ts [hunk 12]
```diff
@@ -1,14 +1,11 @@
 import * as httpm from '../_out';
 import * as path from 'path';
-import * as am from '../_out/auth';
```
Comment (bryanmacfarlane) [inline __tests__/basics.test.ts]: are you going to revert this change?  would like to finish up this PR ...

**Comment by bryanmacfarlane** [comment]
If we're OK with nix only, then only the one comment I just left to revert one line.  Else, we may need more changes if we want windows as devenv

File: __tests__/basics.test.ts (hunk 12 above)
Comment (joshmgross) [reply __tests__/basics.test.ts]: This import isn't used, it was removed completely

Comment (bryanmacfarlane) [reply __tests__/basics.test.ts]: ok
